rp2040-hal/src/uart/reader.rs

```diff
@@ -52,16 +52,37 @@ pub(crate) fn is_readable<D: UartDevice>(device: &D) -> bool {
     device.uartfr.read().rxfe().bit_is_clear()
 }
 
+/// Enable/disable the rx/tx FIFO
+///
+/// Unfortunately, it's not possible to enable/disable rx/tx
+/// independently on this chip
+/// Default is false
+pub fn set_fifos(rb: &RegisterBlock, enable: bool) {
+    if enable {
+        rb.uartlcr_h.modify(|_r, w| w.fen().set_bit())
+    } else {
+        rb.uartlcr_h.modify(|_r, w| w.fen().clear_bit())
+    }
+}
+
+/// Set rx FIFO watermark
+///
+/// See DS: Table 423
+pub fn set_rx_watermark(rb: &RegisterBlock, watermark: FifoWatermark) {
+    let wm = match watermark {
+        FifoWatermark::Bytes4 => 0,
+        FifoWatermark::Bytes8 => 1,
+        FifoWatermark::Bytes16 => 2,
+        FifoWatermark::Bytes24 => 3,
+        FifoWatermark::Bytes28 => 4,
+    };
+    rb.uartifls.modify(|_r, w| unsafe { w.rxiflsel().bits(wm) });
```

Comment:
This unsafe block could be avoided by making the field an enum in the pac.

Reply:
As this requires changes in a different crate first, let's merge it as it is. The pac change can be done separately.